### PR TITLE
Add `reify` expression

### PR DESCRIPTION
**Pull request by magnus-madsen**

### CODE REVIEW

File: main/src/ca/uwaterloo/flix/language/debug/FormatExpression.scala [hunk 1]
```diff
@@ -80,8 +80,7 @@ object FormatExpression {
     case TypedAst.Expression.FixpointFilter(pred, exp, tpe, eff, loc) => s"FixpointFilter($pred, $exp)"
     case TypedAst.Expression.FixpointProjectIn(exp, pred, tpe, eff, loc) => s"FixpointProjectIn($exp, $pred)"
     case TypedAst.Expression.FixpointProjectOut(pred, exp, tpe, eff, loc) => s"FixpointProjectOut($pred, $exp)"
-    case TypedAst.Expression.MatchEff(exp1, exp2, exp3, tpe, eff, loc) => s"MatchEff($exp1, $exp2, $exp3)"
-    case TypedAst.Expression.IfThenElseStar(cond, exp1, exp2, tpe, eff, loc) => s"IfThenElseStar($cond, $exp1, $exp2)"
+    case TypedAst.Expression.Reify(t, tpe, eff, loc) => s"IfThenElseStar($t)"
```
Comment (mlutze) [inline main/src/ca/uwaterloo/flix/language/debug/FormatExpression.scala:83]: `Reify($t)`

File: main/src/ca/uwaterloo/flix/language/phase/Simplifier.scala [hunk 2]
```diff
@@ -307,10 +307,7 @@ object Simplifier extends Phase[TypedAst.Root, SimplifiedAst.Root] {
       case TypedAst.Expression.FixpointProjectOut(_, _, _, _, _) =>
         throw InternalCompilerException(s"Unexpected expression: $exp0.")
 
-      case TypedAst.Expression.MatchEff(_, _, _, _, _, _) =>
-        throw InternalCompilerException(s"Unexpected expression: $exp0.")
-
-      case TypedAst.Expression.IfThenElseStar(_, _, _, _, _, _) =>
+      case TypedAst.Expression.Reify(_, _, _, _) =>
         throw InternalCompilerException(s"Unexpected expression: $exp0.")
```
Comment (mlutze) [inline main/src/ca/uwaterloo/flix/language/phase/Simplifier.scala:311]: This is because the expression is `@EliminatedBy` `Monomorph`, right?

Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/phase/Simplifier.scala:311]: Yes... I wanna say?

File: main/src/ca/uwaterloo/flix/language/debug/FormatExpression.scala (hunk 1 above)
Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/debug/FormatExpression.scala]: Fixed! Thanks!

**Pull request merged**